nomad/structs/structs.go

```diff
@@ -3840,8 +3965,9 @@ func (te *TaskEvent) Copy() *TaskEvent {
 
 func NewTaskEvent(event string) *TaskEvent {
 	return &TaskEvent{
-		Type: event,
-		Time: time.Now().UnixNano(),
+		Type:    event,
+		Time:    time.Now().UnixNano(),
+		Details: make(map[string]string),
```

Comment:
Why is this a map of `type -> details` instead of just being a string containing the details? It doesn't seem like we'd ever have an event with multiple detail entries but maybe I'm missing something?

Reply:
(If we want to keep this a map perhaps we should create a global static map of "Types -> String Keys" instead of hardcoding the strings below?)

Reply:
Created this because we want to get rid of the many fields like ErrorReason/KillReason etc and there can be more than one entry for a single event (e.g kill_signal and kill_reason)

Reply:
Hm if it's dev mode we destroy/GC all allocations on shutdown, so Nomad would be breaking out of that `OUTER` loop in `AllocRunner.Run` and hitting the `AllocRunner.handleDestroy()` code path if that helps.